### DISCUSSION TITLE
Error validating datasource `db`: the URL must start with the protocol `prisma://`

### DISCUSSION THREAD

**Discussion by frostbyteninja** [prisma/prisma#21355] [Q&A] [answered]
### Bug description

PrismaClientKnownRequestError:
Invalid `prisma.users.findUnique()` invocation:

Error validating datasource `db`: the URL must start with the protocol `prisma://`
    at Jn.handleRequestError (C:\Users\shawn\WebstormProjects\discord_listener\node_modules\@prisma\client\runtime\edge.js:36:7072)
    at Jn.handleAndLogRequestError (C:\Users\shawn\WebstormProjects\discord_listener\node_modules\@prisma\client\runtime\edge.js:36:6460)
    at Jn.request (C:\Users\shawn\WebstormProjects\discord_listener\node_modules\@prisma\client\runtime\edge.js:36:6158)
    at async u (C:\Users\shawn\WebstormProjects\discord_listener\node_modules\@prisma\client\runtime\edge.js:41:10106) {
  code: 'P5002',
  clientVersion: '5.3.1',
  meta: undefined
}

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = "BLURRED"
}

model Users {
  UUID String @unique
  UDID String @unique
  Email String @unique @default("notset@example.com")
  DLPass String
  DID Int @unique @id
  DUser String
  DLFileID Json
  DLClaims Json
}

model Guild {
  GID Int @unique @id
  GName String
  GPass String
  GFilter String
  GClaims Json
  GPunishments Json
}

model GuildSettings {
  GID Int @unique @id
  Settings Json
}
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```
"prisma": "^5.4.0"
```

**Comment by janpio** [comment]
@FireTonnor Can you provide some more information please?
Did this project work before you upgrade to Prisma 5.4.0? What database are you trying to talk to? Directly of via Prisma DataProxy/Accelerate?

**Comment by frostbyteninja** [comment]
@janpio No i just started using it today and for the database i am trying to connect to is mysql

**Reply by janpio** [reply]
The error message indicates that you are trying to use the Prisma Client Edge, but that the connection string does not match the requirement of it starting with `prisma://`. Does your connection string start with `mysql://`?
How are you importing Prisma Client in your code?
Are you trying to use Prisma DataProxy or Accelerate?

**Reply by frostbyteninja** [reply]
@janpio 

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = "mysql://mangadiscord:ntLVUqMO]BKPIfU3@BLOCKED:3306/mangadiscord"
}

```

**Reply by janpio** [reply]
Ok, that explains the error.

Did you import `@prisma/client/edge` in your script by accident instead of `@prisma/client`?

**Reply by frostbyteninja** [reply]
@janpio 
```
import {Client, EmbedBuilder, Message} from "discord.js";
import { PrismaClient } from "@prisma//client/edge"
import {v4 as uuidv4} from "uuid";
const prisma = new PrismaClient({datasources: { db: { url: "mysql://mangadiscord:ntLVUqMO]BKPIfU3@BLOCKED:3306/mangadiscord" } }})

```

**Reply by janpio** [reply, marked as answer]
Yes, don't `from "@prisma//client/edge"` but do `from "@prisma/client"` instead and this should work.

**Reply by frostbyteninja** [reply]
Let me try

**Reply by frostbyteninja** [reply]
That fixed it thank you so much

**Reply by cascadingGeek** [reply]
This worked! Thanks a lot

**Reply by DevSadaqat** [reply]
Great. That worked for me as well.

**Reply by mediaformat** [reply]
tried this:
`import { PrismaClient } from "@prisma/client"`
but `"@prisma/client"` does not provide a `PrismaClient` export 🤷 
Otherwise same error as OP
> Error validating datasource db: the URL must start with the protocol prisma://

**Comment by codegeekery** [comment]
Hi, Im having trouble with Invalid `prisma.user.findUnique()` invocation: Error validating datasource `db`: the URL must start with the protocol `prisma://`

im sure that i have.  from "@prisma/client" and still im having that issue the one different is that im using mongoDB url, can help me with this. thanks

Prisma information

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "mongodb"
  url       = env("DATABASE_URL")
}

model Verified {
  id            String   @id @default(auto()) @map("_id") @db.ObjectId
  EmailVerified String
  Active        Boolean
  VerifiedAt    DateTime @default(now())
  User          User?    @relation(fields: [userId], references: [id])
  userId        String?  @db.ObjectId
}
```


my url look like this example `mongodb+srv://example:examplePassword@cluster0.3nc0jrv.mongodb.net/exampleDB`


Prisma Version

`"prisma": "^5.11.0"`

**Comment by OdapX** [comment]


**Reply by janpio** [reply]
Please open a new discussion and provide the full context in your description, thank you.

**Comment by AshishDev-16** [comment]
facing this error :- Error validating datasource `db`: the URL must start with the protocol `prisma://`
but if i try to change the import from  "@prisma//client/edge" to from "@prisma//client" then I get this Error: PrismaClient is not configured to run in Vercel Edge Functions or Edge Middleware. In order to run Prisma Client on edge runtime, either
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters

please help me with this

**Reply by Jauharmuhammed** [reply]
@AshishDev-16  did you find any work around for this?

**Comment by mrymahmadi** [comment]
i using deno v2
how can i import @prisma/client?

**Reply by mediaformat** [reply]
I've tried the methods described [here](https://github.com/prisma/prisma/issues/2452#issuecomment-2491585189), but getting the above error!
